### ISSUE TITLE
Changes in the pymatgen interface and version upgrade

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida-core#3558]
Soon, new versions of pymatgen will change the interface of some of the functionality we provide:
```
  /home/travis/virtualenv/python3.7.1/lib/python3.7/site-packages/aiida/orm/nodes/data/structure.py:907: DeprecationWarning: species_and_occu is deprecated
907  Use site.species instead. This will be deprecated with effect from pymatgen 2020.
908    'symbols': [x.symbol for x in site.species_and_occu.keys()],
```
At the moment we are pinning `pymatgen<=2018.12.12` because it's the last with py2 support.
When we'll drop py2 and we will be able to upgrade, we need to be careful because pymatgen often changes interface, with backward-incompatible changes across years. Probably we can use something like `>2019.1.1,<=2019.12` or similar, but if we go to 2020 we need at least to fix the issue above (and check for other changes they did between 2018 and 2020).

**Comment by sphuber** [comment]
This was already fixed in develop by PR #3480

**Comment by giovannipizzi** [comment]
Oh thanks! I had missed it
